darts/utils/data/training_dataset.py

```diff
@@ -171,6 +167,41 @@ def _memory_indexer(self,
 
         return past_start, past_end, future_start, future_end, cov_start, cov_end
 
+    def _generate_covariates(self,
+                             target: TimeSeries,
+                             covariate: Optional[TimeSeries],
+                             in_range_target: Tuple[int, int],
+                             in_range_cov: Tuple[int, int],
+                             cov_type: CovariateType = CovariateType.NONE):
+        if not self.lazy_encoders.past_encoders and not self.lazy_encoders.future_encoders:
+            return covariate
+
+        past_start, past_end = in_range_target
+        cov_start, cov_end = in_range_cov
+        target = target[past_start:past_end]
+        covariate = covariate[cov_start:cov_end] if covariate is not None else covariate
+
+        if cov_type is CovariateType.PAST:
+            covariate, _ = self.lazy_encoders.encode_train(target=target,
+                                                           past_covariate=covariate,
+                                                           future_covariate=None,
+                                                           encode_future=False)
+        elif cov_type is CovariateType.HISTORIC_FUTURE:
+            _, covariate = self.lazy_encoders.encode_train(target=target,
+                                                           past_covariate=None,
+                                                           future_covariate=covariate,
+                                                           encode_past=False)
+        elif cov_type is CovariateType.FUTURE:
+            _, covariate = self.lazy_encoders.encode_inference(n=self.lazy_encoders.output_chunk_length,
```

Comment:
I find this naming `encode_inference` not ideal, as it is also called for producing training samples.
Perhaps `encode_train` could be renamed into `encode_past` and `encode_inference` into `encode_future` ?

Reply:
`n` is supposed to be the forecast horizon, but here it's set to the `output_chunk_length`. Is this intended?

Reply:
Yes, here it is intended. We want the future part of future covariates on the output chunk of length `output_chunk_length` for the current training sample.

Reply:
as discussed, we can now leave it as it is